src/LibraryManager.Vsix/UI/Controls/Library.xaml.cs

```diff
@@ -114,23 +153,30 @@ private void HandleKeyPress(object sender, KeyEventArgs e)
             }
         }
 
-        private void HandleKeyUp(object sender, KeyEventArgs e)
-        {
-            CaretIndex = LibrarySearchBox.CaretIndex;
-            RefreshSearch();
-        }
-
         private void HandleListBoxKeyPress(object sender, KeyEventArgs e)
         {
             int index = LibrarySearchBox.CaretIndex;
 
             switch (e.Key)
             {
-                case Key.Tab:
-                case Key.Enter:
+                 case Key.Tab:
+                    // SelectedItem could be null if the key press came from keyboard navigation and not commit operation.
+                    // In this case we will just move the focus to next control.
+                    if (SelectedItem == null)
+                    {
+                        LibrarySearchBox.MoveFocus(new TraversalRequest(FocusNavigationDirection.Next));
+                        e.Handled = true;
+                        break;
+                    }
+
                     Commit(SelectedItem);
+                    LibrarySearchBox.Focus();
```

Comment:
>LibrarySearchBox.Focus(); [](start = 20, length = 25)

Same questions as above...

Reply:
Updated this in iteration 8

---
In reply to: [200472453](https://github.com/aspnet/LibraryManager/pull/278#discussion_r200472453) [](ancestors = 200472453)